src/graphics/canvas.rs

```diff
@@ -1,392 +1,85 @@
-                Quat::IDENTITY,
-                glam::vec3(0.0, self.image.height as f32, 0.0),
-            ));
-            crate::graphics::image::draw_image_raw(&ms_canvas.image, ctx, param)?;
-            // restore the old target
-            ctx.gfx_context.data.out = old_target;
-            // and the old shader
-            *ctx.gfx_context.current_shader.borrow_mut() = old_shader;
-        }
-        Ok(())
-    }
 }
 
-impl Drawable for Canvas {
-    fn draw(&self, ctx: &mut Context, param: DrawParam) -> GameResult {
-        self.debug_id.assert(ctx);
-        self.resolve(ctx)?;
-
-        // We have to mess with the scale to make everything
-        // be its-unit-size-in-pixels.
-        let scale_x = param.src.w * f32::from(self.width());
-        let scale_y = param.src.h * f32::from(self.height());
-
-        let param = param.transform(
-            glam::Mat4::from(param.trans.to_bare_matrix())
-                * Matrix4::from_scale(glam::vec3(scale_x, scale_y, 1.0)),
-        );
-
-        let new_param = flip_draw_param_vertical(param);
-
-        crate::graphics::image::draw_image_raw(&self.image, ctx, new_param)
-    }
-    fn dimensions(&self, _: &mut Context) -> Option<Rect> {
-        Some(self.image.dimensions())
-    }
-    fn set_blend_mode(&mut self, mode: Option<BlendMode>) {
-        self.image.blend_mode = mode;
-    }
-    fn blend_mode(&self) -> Option<BlendMode> {
-        self.image.blend_mode
-    }
-}
-
-fn flip_draw_param_vertical(param: DrawParam) -> DrawParam {
-    let param = if let Transform::Matrix(mat) = param.trans {
-        param.transform(
-            glam::Mat4::from(mat)
-                * glam::Mat4::from_scale_rotation_translation(
-                    glam::vec3(1.0, -1.0, 1.0),
-                    Quat::IDENTITY,
-                    glam::vec3(0.0, 1.0, 0.0),
-                ),
-        )
-    } else {
-        panic!("Can not be called with a non-matrix DrawParam");
-    };
-    let new_src = Rect {
-        x: param.src.x,
-        y: (1.0 - param.src.h) - param.src.y,
-        w: param.src.w,
-        h: param.src.h,
-    };
-    param.src(new_src)
-}
-
-/// Set the `Canvas` to render to. Specifying `Option::None` will cause all
-/// rendering to be done directly to the screen.
-pub fn set_canvas(ctx: &mut Context, target: Option<&Canvas>) {
-    match target {
-        Some(surface) => {
-            surface.debug_id.assert(ctx);
-            ctx.gfx_context.data.out = match &surface.ms_canvas {
-                Some(ms_canvas) => ms_canvas.target.clone(),
-                _ => surface.target.clone(),
-            };
-        }
-        None => {
-            ctx.gfx_context.data.out = ctx.gfx_context.screen_render_target.clone();
-        }
-    };
+/// Describes the image load operation when starting a new canvas.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum CanvasLoadOp {
+    /// Keep the existing contents of the image.
+    DontClear,
```

Comment:
DoNotClear is maybe better, Dont sounds too informal

Reply:
i think this language is ok, vulkan for example has `VK_ATTACHMENT_LOAD_OP_DONT_CARE`